### ISSUE TITLE
`quick-xml` incompatible in the future

### ISSUE THREAD

**Issue by wermos** [Keats/gutenberg#2452]
# Bug Report

## Environment

Ubuntu 23.10, with `rustc` version 1.76.0

Zola version:
0.18.0

## Expected Behavior
Tell us what should have happened.
No warnings should be emitted.

## Current Behavior

This is a minor issue, but I thought I'd add it here anyways. Building `zola` from source emits this warning at the end:
```
warning: the following packages contain code that will be rejected by a future version of Rust: quick-xml v0.17.2
```
Since I am not too fluent in Rust, I am not sure how serious the warning is.

## Step to reproduce
Please provide the steps to reproduce the issue.
If the issue is hard to reproduce, please provide a sample repository or sample
that triggers the bug.
Build `zola` from source.

**Comment by Keats** [comment]
It needs to be fixed but I believe the library we are using were planning to fix it last time i checked (months ago).

**Comment by piotr-calus** [comment]
> It needs to be fixed but I believe the library we are using were planning to fix it last time i checked (months ago).

Yes, there is an open PR bumping the minidom version, which will remove the dependency on quick-xml
* AlecTroemel/quickxml_to_serde/pull/26

**Comment by Keats** [comment]
Looks like they might be planning to archive that crate so we should probably move away from it as well.

**Comment by ajbozdar** [comment]
@Keats Following the issue now.

**Comment by jonassmedegaard** [comment]
A simple fix for this issue in now (since February) available: replace the use of crate quickxml_to_serde with [roxmltree_to_serde](https://lib.rs/crates/roxmltree_to_serde)
